app/controllers/clearance/passwords_controller.rb

```diff
@@ -24,10 +27,12 @@ def new
   end
 
   def update
-    @user = find_user_for_update
+    @password_reset = find_password_reset_by_user_id_and_token
+    user = @password_reset.user
 
-    if @user.update_password password_reset_params
-      sign_in @user
+    if user.update_password password_reset_params
+      Clearance::PasswordResetDeactivator.new(user).run
```

Comment:
I did some crazy stuff in https://github.com/thoughtbot/clearance/commit/49ae7ac3d029b2250d37be8ded3b94c5d89a11cb. Let me know what your thoughts are.

I'm finding that validating that the password update is valid kind of hard when wrapped in a transaction...might need some more brainstorming here where we can use `ActiveRecord::Base.transaction` but still be able to fetch useful error messages from either the user or the password reset objects.